src/main/java/org/jenkinsci/plugins/workflow/steps/IsMacStep.java

```diff
@@ -0,0 +1,65 @@
+package org.jenkinsci.plugins.workflow.steps;
+
+import hudson.Extension;
+import hudson.Launcher;
+import hudson.Platform;
+import hudson.remoting.VirtualChannel;
+import jenkins.security.MasterToSlaveCallable;
+
+import java.util.Collections;
+import java.util.Set;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+/**
+ * Checks whether we are running on Mac.
+ */
+public class IsMacStep extends Step {
+
+    @DataBoundConstructor public IsMacStep() {}
+
+    @Override public StepExecution start(StepContext context) throws Exception {
+        return new Execution(context);
+    }
+
+    public static class Execution extends SynchronousNonBlockingStepExecution<Boolean> {
+
+        Execution(StepContext context) {
+            super(context);
+        }
+
+        @Override protected Boolean run() throws Exception {
+            VirtualChannel c = getContext().get(Launcher.class).getChannel();
+            if (c != null)
```

Comment:
I doubt this would be null but i have no idea what to do in cases where this is in fact null. Perhaps it should throw?

Reply:
c would be null when running on Jenkins master.
so it should definitely not throw as the Jenkins master could be running on mac.